src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -0,0 +1,596 @@
+                    for (final ActionArgument executionActionArgument : executionActionArgumentList)
+                    {
+                        if (null != executionActionArgument)
+                        {
+                            if (actionArgument.getName().equals(executionActionArgument.getName()))
+                            {
+                                if (!executionActionArgument.isValueSet())
+                                {
+                                    if (null == badArgument)
+                                    {
+                                        throw new CommandException(
+                                                getCommandUsage(sender, executionTreeNode)
+                                                    + ": no value specified for required argument " + actionArgument.getName());
+                                    }
+                                    else
+                                    {
+                                        throw new CommandException(
+                                                getCommandUsage(sender, executionTreeNode)
+                                                    + ": invalid value '" + badArgument + "' for required argument " + actionArgument.getName());
+                                    }
+                                }
+                                else
+                                {
+                                    foundIt = true;
+                                }
+                            }
+                        }
+                    }
+                }
+                if (!foundIt)
+                {
+                    throw new CommandException(getCommandUsage(sender, executionTreeNode)
+                            + ": missing required parameter " + actionArgument.getName());
+                }
+            }
+        }
+
+        if (null != executionActionArgumentList)
+        {
+            for (final ActionArgument executionActionArgument : executionActionArgumentList)
+            {
+                if (null != executionActionArgument)
+                {
+                    if (!executionActionArgument.isValueSet())
+                    {
+                        throw new CommandException(
+                                getCommandUsage(sender, executionTreeNode)
+                                    + ": invalid value '" + badArgument + "' for required argument " + executionActionArgument.getName());
+                    }
+                }
+            }
+        }
+
+        if (sender instanceof EntityPlayer)
+        {
+            final ForgePermissionNodes forgePermissionNode = actionMenu.getForgePermissionNode();
+
+            final EntityPlayer player = (EntityPlayer) sender;
+            if (!PermissionAPI.hasPermission(player.getGameProfile(), forgePermissionNode.getNodeName(), new PlayerContext(player)))
+            {
+                // TODO: Do something if permission check fails.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)